mlflow/utils/file_utils.py

```diff
@@ -43,9 +44,56 @@
 from mlflow.utils import download_cloud_file_chunk
 from mlflow.utils.request_utils import download_chunk
 
-
 ENCODING = "utf-8"
 MAX_PARALLEL_DOWNLOAD_WORKERS = os.cpu_count() * 2
+_PROGRESS_BAR_DISPLAY_THRESHOLD = 500_000_000  # 500 MB
+
+
+class ArtifactProgressBar:
+    def __init__(self, desc, total, step, enable_pbar=True, **kwargs) -> None:
+        self.total = total
+        self.step = step
+        self.pbar = None
+        self.progress = 0
+
+        if enable_pbar and MLFLOW_ENABLE_ARTIFACTS_PROGRESS_BAR.get():
+            try:
+                from tqdm.auto import tqdm
+
+                self.pbar = tqdm(total=self.total, desc=desc, **kwargs)
+            except ImportError:
+                pass
+
+    @classmethod
+    def chunks(cls, file_size, desc, chunk_size):
+        return cls(
+            desc,
+            total=file_size,
+            step=chunk_size,
+            enable_pbar=file_size >= _PROGRESS_BAR_DISPLAY_THRESHOLD,
+            unit="iB",
+            unit_scale=True,
+            unit_divisor=1024,
+            miniters=1,
+        )
+
+    @classmethod
+    def files(cls, desc, total):
+        return cls(desc, total=total, step=1)
+
+    def update(self):
+        if self.pbar:
+            update_step = min(self.total - self.progress, self.step)
+            self.pbar.update(update_step)
+            self.pbar.refresh()
+            self.progress += update_step
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        if self.pbar:
+            self.pbar.close()
```

Comment:
/ai I want to remove the `enable_pbar` argument in this code. Please propose a fix.